File: android/src/main/java/com/reactnativesolutionsgyrohandler/GyroHandlerModule.java
package com.reactnativesolutionsgyrohandler;

import androidx.annotation.Nullable;

import com.facebook.react.common.MapBuilder;
import com.facebook.react.uimanager.SimpleViewManager;
import com.facebook.react.uimanager.ViewGroupManager;
import com.facebook.react.uimanager.ThemedReactContext;
import java.util.Map;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.views.scroll.ScrollEventType;

public class GyroHandlerModule extends SimpleViewManager<RNGyroHandler>  {

	private ReactApplicationContext mReactContext;

	public GyroHandlerModule(ReactApplicationContext reactContext) {
		this.mReactContext = reactContext;
	}

	@Override
	public @Nullable Map getExportedCustomDirectEventTypeConstants() {
		return MapBuilder.<String, Object>builder()
			.put(
				"onGyroChange",
				MapBuilder.of("registrationName", "onGyroChange")
			).build();
	}

	@Override
	public String getName() {
			return "RNGyroHandler";
	}

	protected RNGyroHandler createViewInstance(ThemedReactContext reactContext) {
		return new RNGyroHandler(reactContext, this.mReactContext);
	}
}

File: android/src/main/java/com/reactnativesolutionsgyrohandler/GyroHandlerModule.kt
package com.reactnativesolutionsgyrohandler

import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod
import com.facebook.react.bridge.Promise

class GyroHandlerModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(reactContext) {

    override fun getName(): String {
        return "GyroHandler"
    }

    // Example method
    // See https://facebook.github.io/react-native/docs/native-modules-android
    @ReactMethod
    fun getDeviceName(promise: Promise) {
        promise.resolve(android.os.Build.MODEL)
    }
}
